src/Auth0.Core/Exceptions/ErrorApiException.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Net;
+
+namespace Auth0.Core.Exceptions
+{
+    /// <summary>
+    /// Represents errors that occur when making API calls.
+    /// </summary>
+    public class ErrorApiException : ApiException
+    {
+        /// <summary>
+        /// Optional <see cref="Core.ApiError"/> from the failing API call.
+        /// </summary>
+        public ApiError ApiError { get; }
+
+        /// <summary>
+        /// <see cref="HttpStatusCode"/> code from the failing API call.
+        /// </summary>
+        public HttpStatusCode StatusCode { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ErrorApiException"/> class.
+        /// </summary>
+        public ErrorApiException()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ErrorApiException"/> class with a specified error message.
+        /// </summary>
+        /// <param name="message">The error message that explains the reason for the exception.</param>
+        public ErrorApiException(string message)
+            : base(message)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ErrorApiException"/> class with a specified error message
+        /// and a reference to the inner exception that is the cause of this exception.
+        /// </summary>
+        /// <param name="message">The error message that explains the reason for the exception.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception, or a null 
```

Comment:
**Question:** Can you explain an "inner" exception? Like a child of the main exception?

Reply:
Yeah it's what exception caused this exception. These are just standard Exception constructors every constructor in .net should have.